transport/pool.go

```diff
@@ -34,30 +33,34 @@ func NewConnPool(addr string, connConfig ConnConfig) (*ConnPool, error) {
 	return &r.pool, nil
 }
 
-func (p *ConnPool) RandConn() *Conn {
-	conn := p.loadConn(rand.Intn(p.nrShards))
-	if conn != nil {
-		return conn
-	}
-	return p.NextConn()
-}
+func (p *ConnPool) LeastBusyConn() *Conn {
+	var (
+		leastBusyConn *Conn
+		minBusy       = math.MaxUint32
+	)
 
-func (p *ConnPool) NextConn() *Conn {
 	idx := p.idx.Add(1)
 	for i := 0; i < p.nrShards; i++ {
 		if conn := p.loadConn(int((idx + uint64(i)) % uint64(p.nrShards))); conn != nil {
-			return conn
+			if waiting := conn.Waiting(); waiting < minBusy {
```

Comment:
Please correct me if I'm wrong, but I'm afraid maxuint32 might be negative on 32bit OS. I would go for uint32 as the type for minBusy and Waiting().